pkg/controller/container-runtime-config/helpers.go

```diff
@@ -431,14 +435,18 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
 	}
 
 	ctrcfg := cfg.Spec.ContainerRuntimeConfig
-	if ctrcfg.PidsLimit > 0 && ctrcfg.PidsLimit < minPidsLimit {
-		return fmt.Errorf("invalid PidsLimit %q, cannot be less than 20", ctrcfg.PidsLimit)
+	if ctrcfg.PidsLimit != nil && *ctrcfg.PidsLimit > 0 && *ctrcfg.PidsLimit < minPidsLimit {
+		return fmt.Errorf("invalid PidsLimit %q, cannot be less than 20", *ctrcfg.PidsLimit)
```

Comment:
this error message doesn't match what we're verifying anymore

Reply:
minPidsLimit is 20. I am not sure why this value is checked. Should it be dropped?

Reply:
apparently I can't read. No that's fine :)

Reply:
The minimum is 20 because anything less than 20 will most definitely not be enough pids for an application, so we set this limit to stop the user from hitting this issue. However, setting it to 0 sets it to the max value of pids allowed.

Reply:
So should this say “if positive, cannot be less than 20”, to be pedantic?

(This is not the purpose of this PR, but given the demonstrated confusion, it would be nice to clean that up to say `"%d", minPidsLimit` instead of hard-coding the string.)